### PR TITLE
Fix network ordering & allow assigning DHCP IP to a server

### PR DESCRIPTION
**Pull request by nvthongswansea**
Ref #186 & #142. Changes:
- The field `ordering` field is DEPRECATED. It will have no effect. Instead, the network ordering of a server corresponds to the order of the networks in that server resource block.
- Upgrade gsclient-go v3.8.0.
- Updating server-storage relations and server-network relations' properties (except the order) will no longer require the server to be shutdown.
- Allow assigning DHCP IP to a server.
- Update server resource docs.
- Make field `dhcp_range` computed (More info [commit df3173e](https://github.com/gridscale/terraform-provider-gridscale/pull/190/commits/df3173ea86b5b87fbb34301c4db365cf08accbf4))

NOTE:
- Since the field `ordering` is deprecated, the field will be ignored, even if the old server blocks still use it. In case that the old server blocks don't use that field, doing `terraform apply` for the first time after upgrading to the new `terraform-provider-gridscale` will restart the multiple stack servers. The further `terraform apply` shouldn't restart the servers if there are no changes in servers' network ordering or there are no networks being attached/detached to/from the servers.

### CODE REVIEW

File: gridscale/resource_gridscale_server.go [hunk 1]
```diff
@@ -758,13 +756,24 @@ func resourceGridscaleServerUpdate(d *schema.ResourceData, meta interface{}) err
 		if err != nil {
 			return fmt.Errorf("%s error: %v", errorPrefix, err)
 		}
+
+		//Update relationship between the server and networks
+		err = serverDepClient.UpdateNetworksRel(ctxWTimeout)
```
Comment (itakouna) [inline gridscale/resource_gridscale_server.go]: I see this function is used two times: one with `shutdown` and the other one without `shutdown`. Would you explain the difference?

Comment (nvthongswansea) [reply gridscale/resource_gridscale_server.go]: Nice find. I will check this dup.

File: gridscale/resource_gridscale_server.go [hunk 2]
```diff
@@ -191,6 +191,7 @@ func resourceGridscaleServer() *schema.Resource {
 						"ordering": {
 							Type:       schema.TypeInt,
 							Optional:   true,
```
Comment (itakouna) [inline gridscale/resource_gridscale_server.go:203]: Do we still need the `Optional` attribute with `Computed`?

Comment (nvthongswansea) [reply gridscale/resource_gridscale_server.go:203]: > Do we still need the `Optional` attribute with `Computed`?

Yes. If you remove the `optional` attribute, the backward compatibility is broken (if some users already set that field `ordering`)

**Review by itakouna — APPROVED** [review]

**Pull request merged**